spring-integration-jms/src/main/java/org/springframework/integration/jms/dsl/JmsInboundGateway.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jms.dsl;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.integration.context.OrderlyShutdownCapable;
+import org.springframework.integration.gateway.MessagingGatewaySupport;
+import org.springframework.integration.jms.ChannelPublishingJmsMessageListener;
+import org.springframework.integration.jms.JmsMessageDrivenEndpoint;
+import org.springframework.jms.listener.AbstractMessageListenerContainer;
+import org.springframework.messaging.MessageChannel;
+
+/**
+ * A wrapper around the {@link JmsMessageDrivenEndpoint} to work around
+ * Java single inheritance.
+ *
+ * @author Artem Bilan
+ * @author Gary Russell
+ * @since 5.0
+ */
+public class JmsInboundGateway extends MessagingGatewaySupport implements DisposableBean, OrderlyShutdownCapable {
```

Comment:
Maybe this class can go out from `dsl` package?
It is fully makes sense to live in the regular `inbound` package.
Plus it is valuable to to be used as a standalone component.